lib/kubernetes-deploy/restart_task.rb

```diff
@@ -32,7 +37,10 @@ def perform(deployments_names = nil)
       @logger.phase_heading("Initializing restart")
       verify_namespace
       deployments = identify_target_deployments(deployments_names)
-
+      if server_version < Gem::Version.new(MIN_KUBE_VERSION)
+        @logger.warn("Minimum cluster version requirement of #{MIN_KUBE_VERSION} not met. "\
+        "Using #{server_version} could result in unexpected behavior as it is no longer tested against")
```

Comment:
You could make a common method (maybe `KubernetesDeploy::Errors.server_version_warning(server_version)`?) that all three tasks could call to avoid repeating the full string everywhere.